psycopg/psycopg/connection.py

```diff
@@ -534,8 +551,33 @@ def _rollback_gen(self) -> PQGen[None]:
     def _fetch_many_gen(self) -> PQGen[List["PGresult"]]:
         """Return results of one query from the database."""
         results = yield from fetch_many(self.pgconn)
+        logger.debug(
+            "received '%s' result(s) from pipeline",
+            ", ".join(pq.ExecStatus(r.status).name for r in results),
+        )
         return results
 
+    def _end_pipeline_gen(self) -> PQGen[None]:
+        """Empty the pipeline queue from unfetched results, usually those from
+        commands (e.g. COMMIT) or pipeline synchronization points which
+        results are not expected to be fetched by the user.
+        """
+        while self._pipeline_queue:
+            queued = self._pipeline_queue.popleft()
+
+            if queued is not None:
```

Comment:
Is it really an error condition to have unfetched results? What if the user doesn't care (e.g. UPDATE or SELECT for a side effect)? I could also see race conditions in case the cursor is managed by a thread while the pipeline block is managed by the main thread (which sounds like a likely way to use it).